### DISCUSSION TITLE
Duplicate messages anomaly?

### DISCUSSION THREAD

**Discussion by mihalt** [Terrance/SkPy#235] [Q&A]
I found that any of 2 my messages absolutely the same, except id and version. In skype graghical app is only one message. 

I understand, that it is not skpy issue. I would like only to know your opinion about skype web api behaviour. 

I assume, that skype is sending second message, if it will not get by recipient after some time? due to internet connection maybe? 
So, can I only delete messages like this freely?

**Comment by Terrance** [comment]
I can't quite parse what you're saying -- SkPy and/or Skype for Web (where are you seeing the duplicates -- in `getMsgs()`? in the web UI?) are duplicating some (all? to specific contacts?) messages you're sending(?), but the desktop/mobile clients are showing it once as expected?

> absolutely the same, except id and version

Which ID?  Are you looking at an edit of a message?

> skype is sending second message, if it will not get by recipient after some time

Unlikely, given it doesn't show in all clients, and messages are stored in the cloud so can be retrieved whenever.

**Reply by mihalt** [reply]
I give you example of messages. They are from getMsgs(), correct. For current time I saw duplicates like that mainly from that contact. He was sender. A bit not understood about desktop/mobile, but graphical interface of skype showing only one message, correct. 
![изображение](https://user-images.githubusercontent.com/30621622/232871516-d1c36cad-8526-4df4-bc72-41d8f38c2d95.png)

and  second message
![изображение](https://user-images.githubusercontent.com/30621622/232870525-cf644d1f-423e-46e3-a0b8-988ee47853a0.png)

**Reply by Terrance** [reply]
> A bit not understood about desktop/mobile, but graphical interface of skype

Which graphical interface?  Skype for Web?

Check the network requests there and see if the message is duplicated there too (i.e. if the client is dealing with it somehow).
